src/components/__tests__/HabitForm.test.tsx

```diff
@@ -1,25 +1,24 @@
 import React from "react";
-import Form from "../Form";
+import HabitForm from "../HabitForm";
 import renderer from "react-test-renderer";
 import { testReducer } from "../../reducers/__tests__/index.test";
 import { createStore } from "redux";
 import { Provider } from "react-redux";
+import { Route, RouteComponentProps, StaticRouter } from "react-router-dom";
 
 const store = createStore(testReducer);
 
 it("renders the form in edit mode correctly", () => {
   const tree = renderer
     .create(
       <Provider store={store}>
-        <Form
-          now={new Date("02/04/2020")}
-          match={{
-            path: "/:id/edit",
-            url: "/0/edit",
-            isExact: true,
-            params: { id: "0" }
-          }}
-        />
+        <StaticRouter>
+          <Route
+            render={(props: RouteComponentProps) => (
+              <HabitForm {...props} now={new Date("02/04/2020")} id="0" />
```

Comment:
We should pull out the `new Date(...)` to the top level instead of being within the render function -- it's a static value, after all. Instead, we'd do:

```js
const store = ...
const now = new Date("02/04/2020");
```